File: 5f9aba12-2f21-4ed8-8aa6-9283a9ab9fe0/src/main/java/com/sportsevent/dto/ApiResponse.java
package com.sportsevent.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ApiResponse<T> {

    private int code;
    private String message;
    private T data;
    private LocalDateTime timestamp = LocalDateTime.now();
    private boolean success;

    public static <T> ApiResponse<T> success(T data) {
        ApiResponse<T> response = new ApiResponse<>();
        response.setCode(200);
        response.setSuccess(true);
        response.setMessage("Success");
        response.setData(data);
        return response;
    }

    public static <T> ApiResponse<T> success(String message, T data) {
        ApiResponse<T> response = success(data);
        response.setMessage(message);
        return response;
    }

    public static <T> ApiResponse<T> warning(String message, T data) {
        ApiResponse<T> response = new ApiResponse<>();
        response.setCode(299);
        response.setSuccess(true);
        response.setMessage(message);
        response.setData(data);
        return response;
    }

    public static <T> ApiResponse<T> error(int code, String message) {
        ApiResponse<T> response = new ApiResponse<>();
        response.setCode(code);
        response.setSuccess(false);
        response.setMessage(message);
        return response;
    }

    public static <T> ApiResponse<T> badRequest(String message) {
        return error(400, message);
    }

    public static <T> ApiResponse<T> notFound(String message) {
        return error(404, message);
    }

    public static <T> ApiResponse<T> serverError(String message) {
        return error(500, message);
    }
}

File: 9096d8e2-d12b-468a-91ae-53c97e7bf587/backend/src/main/kotlin/com/ems/dispatch/dto/ApiResponse.kt
package com.ems.dispatch.dto

import java.time.LocalDateTime

data class ApiResponse<T>(
    val code: Int,
    val message: String,
    val data: T? = null,
    val timestamp: LocalDateTime = LocalDateTime.now(),
    val success: Boolean = code == 200
) {
    companion object {
        fun <T> success(data: T? = null, message: String = "Success"): ApiResponse<T> {
            return ApiResponse(200, message, data)
        }

        fun <T> error(code: Int = 500, message: String = "Internal Server Error"): ApiResponse<T> {
            return ApiResponse(code, message, null)
        }

        fun <T> badRequest(message: String = "Bad Request"): ApiResponse<T> {
            return ApiResponse(400, message, null)
        }

        fun <T> notFound(message: String = "Not Found"): ApiResponse<T> {
            return ApiResponse(404, message, null)
        }

        fun <T> unauthorized(message: String = "Unauthorized"): ApiResponse<T> {
            return ApiResponse(401, message, null)
        }

        fun <T> forbidden(message: String = "Forbidden"): ApiResponse<T> {
            return ApiResponse(403, message, null)
        }
    }
}

data class PageResponse<T>(
    val content: List<T>,
    val page: Int,
    val size: Int,
    val totalElements: Long,
    val totalPages: Int,
    val hasNext: Boolean,
    val hasPrevious: Boolean
)
